impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -84,7 +84,7 @@ public void write(XmlWriterRequest<PluginDescriptor> request) throws XmlWriterEx
         OutputStream outputStream = request.getOutputStream();
         Writer writer = request.getWriter();
         if (writer == null && outputStream == null && path == null) {
-            throw new IllegalArgumentException("writer, outputStream or path must be non null");
+            throw new IllegalArgumentException("writer, output stream, or path must be non null");
```

Comment:
non-null (add hyphen)

Reply:
Thanks for guidance.

FYI: Suggestion

its a nice feature use on your wish. I will follow ether way.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/08e5b561-56cf-4394-970b-b53470ae38ea" />